src/components/timecard/edit-shift/EditShift.jsx

```diff
@@ -0,0 +1,509 @@
+      return false;
+    }
+    let newErrors = [];
+    let errorsHandled = true;
+
+    for (const error of errors) {
+      if (error.field === clashingProperties.startAndEndTime) {
+        setClashingTimes(error.data);
+        setClashingProperty(error.field);
+        newErrors.push({
+          key: 'bothDatesOverlapping',
+          inputName: 'shift-start-time',
+          message:
+            'Your start and finish times must not overlap with another time period',
+          errorPriority: 1,
+        });
+      } else if (error.field === clashingProperties.startTime) {
+        setClashingTimes(error.data);
+        setClashingProperty(error.field);
+        newErrors.push({
+          key: 'overlappingStart',
+          inputName: 'shift-start-time',
+          message: 'Your start time must not overlap with another time period',
+          errorPriority: 1,
+        });
+      } else if (error.field === clashingProperties.endTime) {
+        setClashingTimes(error.data);
+        setClashingProperty(error.field);
+        newErrors.push({
+          key: 'overlappingFinish',
+          inputName: 'shift-finish-time',
+          message: 'Your finish time must not overlap with another time period',
+          errorPriority: 2,
+        });
+      } else {
+        errorsHandled = false;
+        break;
+      }
+    }
+    if (errorsHandled) {
+      setSummaryErrors(newErrors);
+      return true;
+    }
+
+    return false;
+  };
+
+  const onSubmit = async (formData) => {
+    setSummaryErrors([]);
+    const validatedData = validateSubmittedData(formData);
+
+    if (validatedData.isValid) {
+      const timecardPayload = {
+        ownerId: userId,
+        timePeriodTypeId: timeEntry.timePeriodTypeId,
+        actualStartTime: validatedData.startDateTime,
+        actualEndTime: validatedData.finishDateTime,
+      };
+
+      validateServiceErrors(
+        setServiceError,
+        async () => {
+          const params = new UrlSearchParamBuilder()
+            .setTenantId('00000000-0000-0000-0000-000000000000')
+            .getUrlSearchParams();
+
+          const response = !timeEntry.timeEntryId
+            ? await createTimeEntry(timecardPayload, params)
+            : await updateTimeEntry(
+                timeEntry.timeEntryId,
+                timecardPayload,
+                params
+              );
+          if (response.status === 200) {
+            //to be made better ^^
+            if (
+              startEntryExists &&
+              hasShiftMovedFromTimecard(
+                validatedData.startDateTime,
+                validatedData.finishDateTime
+              )
+            ) {
+              setMessages(
+                validatedData.startDateTime,
+                validatedData.finishDateTime
+              );
+              removeTimecardContextEntry(
+                timeEntries,
+                setTimeEntries,
+                timeEntriesIndex
```

Comment:
We don't have timecard context anymore - should this be removed everywhere it's used?

Reply:
Renamed all usages to a more relevant name